File: packages/emitter/testcases/common/classes/generic-methods/MethodInNonGenericClass.ts
import { int } from "@tsonic/core/types.js";

export interface Wrapper<T> {
  value: T;
}

export interface Pair<K, V> {
  key: K;
  value: V;
}

export class Utils {
  // Static generic method
  static identity<T>(value: T): T {
    return value;
  }

  // Instance generic method
  wrap<T>(value: T): Wrapper<T> {
    return { value };
  }

  // Generic method with multiple params
  pair<K, V>(key: K, value: V): Pair<K, V> {
    return { key, value };
  }
}

File: packages/emitter/testcases/common/expected/classes/generic-methods/MethodInNonGenericClass.cs
namespace TestCases.common.classes.genericmethods
{
    public class Wrapper<T>
    {
        public required T value { get; set; }
    }

    public class Pair<K, V>
    {
        public required K key { get; set; }

        public required V value { get; set; }
    }

    public class Utils
    {
        public static T identity<T>(T value)
        {
            return value;
        }

        public Wrapper<T> wrap<T>(T value)
        {
            return new Wrapper<T> { value = value };
        }

        public Pair<K, V> pair<K, V>(K key, V value)
        {
            return new Pair<K, V> { key = key, value = value };
        }
    }
}